/***************************************************************************
 *
 * Copyright (c) 2014 Baidu, Inc. All Rights Reserved.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 *
 **************************************************************************/
// Author: Zhang Yuncong
//
// Pipeline Configs proto


package baidu.bigflow;

import "flume/proto/config.proto";

message PbPipeline {
    enum Type {
        TEST    = 1;
        LOCAL   = 2;
        DAGMR   = 3;
        SPARK   = 4;
        TM      = 5;
        MONITOR = 100;
    }

    required Type type = 1 [default = TEST];
    optional baidu.flume.PbLocalConfig local_config = 2;
    optional baidu.flume.PbMapReduceConfig mr_config = 3;
    optional baidu.flume.PbSparkConfig spark_config = 4;
    optional baidu.flume.PbTMConfig tm_config = 5;
}
